mcgill-augmented-reality-backend/src/main/java/mcgillar/backend/controllers/user/AppUserController.java

```diff
@@ -67,4 +69,22 @@ public ResponseEntity<ValidatedUser> changeUsername(@RequestParam String newUser
         return new ResponseEntity<ValidatedUser>(user, HttpStatus.ACCEPTED);
     }
 
+    /**
+     * 
+     * @param authentication
+     * @return deleted user
+     * 
+     * !! Danger !!
+     */
+    @DeleteMapping("delete-account")
+    public ResponseEntity<ValidatedUser> deleteAccount(
+        Authentication authentication
+    ) {
+        String username = authentication.getName();
+        AppUser user = appUserService.getUserByUsername(username);
+        ValidatedUser to = ValidatedUser.getInstance(user);
+        appUserService.deleteAccount(username);
+        return new ResponseEntity<ValidatedUser>(to, HttpStatus.ACCEPTED);
```

Comment:
Should we be returning the user that is now deleted? what are doing with that info? Should we instead just return a flag to indicate success?

Reply:
Ok will fix!